hoomd/hpmc/pytest/test_shape.py

```diff
@@ -0,0 +1,109 @@
+                elif key != 'shapes':
+                    assert test_dict[key] == val
+
+
+def test_default_args(integrator_args):
+    for shape_integrator, valid_args, invalid_args in integrator_args():
+        mc = shape_integrator(23456)
+        mc.shape['A'] = dict()
+        for key, val in list(mc.shape["A"].items()):
+            assert mc.shape["A"][key] == val
+
+
+def test_shape_params(integrator_args):
+    for shape_integrator, valid_args, invalid_args in integrator_args():
+        mc = shape_integrator(23456)
+        for args in valid_args:
+            mc.shape["A"] = args
+            for key, val in list(args.items()):
+                if key == "vertices":
+                    np.testing.assert_allclose(mc.shape["A"][key], val)
+                else:
+                    assert mc.shape["A"][key] == val
+        for args in invalid_args:
+            tf = False
+            try:
+                mc.shape["A"] = args
+            except:
+                tf = True
+            assert tf
+
+
+def test_shape_attached(dummy_simulation_factory, integrator_args):
+    for shape_integrator, valid_args, invalid_args in integrator_args():
+        mc = shape_integrator(23456)
+        for args in valid_args:
+            mc.shape["A"] = args
+            sim = dummy_simulation_factory()
+            sim.operations.add(mc)
+            sim.operations.schedule()
+            for key, val in list(args.items()):
+                if isinstance(mc.shape["A"][key], list) and len(mc.shape["A"][key]) > 0 \
+                and key != 'shapes':
+                    np.testing.assert_allclose(mc.shape["A"][key], val)
+                elif key != 'shapes':
+                    assert mc.shape["A"][key] == val
+
+
+def test_overlaps(device, lattice_simulation_factory, integrator_args):
+    for shape_integrator, valid_args, invalid_args in integrator_args():
+        if 'union' not in str(shape_integrator).lower():
+            args = valid_args[0]
+            mc = shape_integrator(23456)
+            mc.shape['A'] = args
+
+            sim = lattice_simulation_factory(dimensions=2, n=(2, 1), a=0.25)
+            sim.operations.add(mc)
+
+            sim.operations.schedule()
+            sim.run(1)
+            assert mc.overlaps > 0
+
+            s = sim.state.snapshot
+            if s.exists:
+                s.particles.position[0] = (0, 0, 0)
+                s.particles.position[1] = (0, 8, 0)
+            sim.state.snapshot = s
+            assert mc.overlaps == 0
+
+            s = sim.state.snapshot
+            if s.exists:
+                s.particles.position[0] = (0, 0, 0)
+                s.particles.position[1] = (0, 0.5, 0)
+            sim.state.snapshot = s
+            assert mc.overlaps == 1
+
+
+def test_moves(device, lattice_simulation_factory, integrator_args):
+    dims = 3
+    for shape_integrator, valid_args, invalid_args in integrator_args():
+        if 'union' not in str(shape_integrator).lower():
```

Comment:
Do we do different testing for unions?

Reply:
I am currently getting a KeyError whenever I try to attach a union

Reply:
Fixed fe04009df9